engine/table/src/main/java/io/deephaven/engine/table/impl/util/JobScheduler.java

```diff
@@ -129,13 +130,15 @@ private static void onUnexpectedJobError(@NotNull final Exception exception) {
                 final int count,
                 @NotNull final IterateResumeAction<CONTEXT_TYPE> action,
                 @NotNull final Runnable onComplete,
+                @NotNull final Runnable onCompleteCleanup,
```

Comment:
Majority of current invocations are: `() -> {}`.  Can we consider an overload with this as default (where no cleanup is needed).

Reply:
I'm not a fan of overloading in general. If we were going to do an overload, I might choose to have a more-explicit observer interface (like StreamObserver) instead of 3 separate callbacks; or choose to do a builder-style construction.